app/javascript/packages/one-time-code-input/index.js

```diff
@@ -19,10 +19,10 @@ class OneTimeCodeInput {
   }
 
   createHiddenInput() {
-    const { input, form } = this.elements;
+    const { input } = this.elements;
     const hiddenInput = /** @type {HTMLInputElement} */ (document.createElement('input'));
-    const label = form?.querySelector(`label[for="${input.id}"]`);
-    const modifiedId = `${input.id}-${Math.floor(Math.random() * 1000000)}`;
+    const label = input.ownerDocument.querySelector(`label[for="${input.id}"]`);
+    const modifiedId = `${input.id}-${Math.floor(Math.random() * 1000000)} `;
```

Comment:
this is changed back to make the tests fail on purpose right?

Reply:
Oof, yeah, I'd temporarily changed it back. I was unsuccessful in finding a test case to make it fail, however.

Reply:
```suggestion
    const modifiedId = `${input.id}-${Math.floor(Math.random() * 1000000)}`;
```